test/e2e/storage/sanity.go

```diff
@@ -995,3 +997,46 @@ func WaitForPodsWithLabelRunningReady(c clientset.Interface, ns string, label la
 		})
 	return pods, err
 }
+
+// Register list of volumes before test, using out-of-band host commands (i.e. not CSI API).
+// TODO: remove debug prints before merging
+func GetHostVolumes() map[string][]string {
+	var cmd string
+	var hdr string
+	switch os.Getenv("TEST_DEVICEMODE") {
+	case "lvm":
+		// lvs adds many space (0x20) chars at end, we could squeeze
+		// repetitions using tr here, but TrimSpace() below strips those away
+		cmd = "sudo lvs --foreign --noheadings"
+		hdr = "LVM Volumes"
+	case "direct":
+		// ndctl produces multiline block. We want one line per namespace.
+		// Remove double quotes, delete lines dev:xyz and blockdev:xyz as these elems may change after reboot,
+		// remove newlines, then insert one at the end, clean some more.
+		cmd = "sudo ndctl list |tr -d '\"' |grep -v '^    dev:' |grep	-v '^    blockdev:' |tr -d '\n' |tr ']' '\n' |tr -d '[{}' |tr -d ' '"
+		hdr = "Namespaces"
+	}
+	result := make(map[string][]string)
+	// Instead of trying to find out number of hosts, we trust the set of
+	// ssh.N helper scripts matches running hosts, which should be the case in
+	// correctly running tester system. We run ssh.N commands until a ssh.N
+	// script appears to be "no such file". Have some quite-high upper
+	// limit for loop "just in case".
+	for worker := 1; worker < 1024; worker++ {
```

Comment:
Can you leave out this artificial upper limit? Just start at 1 and run until there's no corresponding ssh script.

Reply:
yes, can do. I added it with idea to not become looking as potential forever loop. But in practice, it will not be, so we can leave out indeed. Having a upper limit introduces another artificial limit of max num of workers supported here.

Reply:
removed upper limit checking and force-pushed